### ISSUE TITLE
java.lang.NullPointerException: Must have Authentication.

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#279]
```
NullPointerException: Must have Authentication.
java.lang.NullPointerException: Must have Authentication.
    at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:907)
    at ubic.gemma.core.job.executor.webapp.TaskRunningServiceImpl.checkTaskCommand(TaskRunningServiceImpl.java:182)
    at ubic.gemma.core.job.executor.webapp.TaskRunningServiceImpl.submitLocalTask(TaskRunningServiceImpl.java:101)
    at ubic.gemma.core.job.executor.webapp.TaskRunningServiceImpl.submitLocalTask(TaskRunningServiceImpl.java:93)
    at ubic.gemma.core.job.executor.webapp.TaskRunningServiceImpl.submitRemoteTask(TaskRunningServiceImpl.java:170)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:166)
    at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204)
    at com.sun.proxy.$Proxy960.submitRemoteTask(Unknown Source)
    at ubic.gemma.core.search.indexer.IndexServiceImpl$IndexerJob.execute(IndexServiceImpl.java:214)
    at ubic.gemma.core.search.indexer.IndexServiceImpl$IndexerJob.execute(IndexServiceImpl.java:205)
    at ubic.gemma.core.job.executor.common.ExecutingTask.call(ExecutingTask.java:55)
    at ubic.gemma.core.job.executor.common.ExecutingTask.call(ExecutingTask.java:29)
    at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
    at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
    at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
```

**Comment by arteymix** [comment]
This can be fixed by using the callable/runnable fix we introduced in e0e1b41718c152f278efa6f3ab5b5acd4371f987

**Comment by arteymix** [comment]
We can remove this check now because we no longer rely on the `Authentication` object being non-null, only the security context.
